lib/tree-view.coffee

```diff
@@ -671,6 +676,24 @@ class TreeView extends View
   toggleSide: ->
     toggleConfig('tree-view.showOnRightSide')
 
+  moveEntry: (initialPath, newDirectoryPath) ->
+    if initialPath is newDirectoryPath
+      return
+
+    entryName = path.basename(initialPath)
+    newPath = "#{newDirectoryPath}/#{entryName}".replace(/\s+$/, '')
+
+    try
+      fs.makeTreeSync(newDirectoryPath) unless fs.existsSync(newDirectoryPath)
+      fs.moveSync(initialPath, newPath)
+
+      if repo = repoForPath(newPath)
+        repo.getPathStatus(initialPath)
+        repo.getPathStatus(newPath)
+
+    catch error
+      console.warn("#{error.message}.")
```

Comment:
This should post a notification similar to [this code](https://github.com/atom/tree-view/blob/fc3c32853735d708dfb8d91e5647992e76349b75/lib/tree-view.coffee#L569) rather than just emitting to the console. I don't think that's visible enough.